infer/SigSuggestion.cc

```diff
@@ -56,16 +56,34 @@ optional<core::AutocorrectSuggestion::Edit> maybeSuggestExtendTSig(core::Context
     return core::AutocorrectSuggestion::Edit{nextLineLoc.value(), fmt::format("{}extend T::Sig\n", prefix)};
 }
 
-core::TypePtr extractArgType(core::Context ctx, cfg::Send &send, core::DispatchComponent &component, int argId) {
+core::TypePtr extractArgType(core::Context ctx, cfg::Send &send, core::DispatchComponent &component,
+                             optional<core::NameRef> keyword, int argId) {
     ENFORCE(component.method.exists());
     const auto &args = component.method.data(ctx)->arguments();
     if (argId >= args.size()) {
         return nullptr;
     }
-    const auto &to = args[argId].type;
+
+    core::TypePtr to = nullptr;
+    if (keyword.has_value()) {
+        auto name = *keyword;
+        for (auto &argInfo : args) {
+            if (argInfo.flags.isKeyword && argInfo.name == name) {
+                to = argInfo.type;
+                break;
+            }
+        }
+        if (to == nullptr) {
+            return nullptr;
+        }
+    } else {
+        to = args[argId].type;
+    }
```

Comment:
We were only considering positional arguments before as sends previously took all the keyword arguments as a single positional hash. With inlined keyword arguments we can make use of the symbols we parse out of the send.